regression/strings-smoke-tests/java_parselong_5/test_parselong.java

```diff
@@ -2,10 +2,11 @@
 {
     public static void main(String[] args)
     {
-        // -2^35
-        String str = new String("-100000000000000000000000000000000000");
+        // -(2^63 - 1) - note that -2^63 should work but doesn't because of a
+        // limitation in the current code
```

Comment:
java_parseint_with_radix_knownbug is for -2^31 in binary as in int, which has the same problem. Anything which fixes that and doesn't do too much special casing would also fix -2^63 in binary as a long. Let me know if you'd like me to make a KNOWNBUG test for -2^63 in binary as a long as well.

Btw, the problem is that the string is so long that we want to reduce i modulo radix^n, but n is so large that radix^n overflows, e.g. n is 31 for an int or 63 for a long. I guess (-radix)^n wouldn't overflow, but it would make other things more complicated.